src/code/z_play.c

```diff
@@ -1,81 +1,1608 @@
 #include "z64quake.h"
 #include "z64shrink_window.h"
 #include "z64view.h"
+#include "overlays/gamestates/ovl_daytelop/z_daytelop.h"
+#include "overlays/gamestates/ovl_opening/z_opening.h"
+#include "overlays/gamestates/ovl_file_choose/z_file_choose.h"
+#include "overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_scope.h"
+
+s32 gDbgCamEnabled = false;
+u8 D_801D0D54 = false;
+
+// bss
+s16 sTransitionFillTimer;
+Input D_801F6C18;
+FbDemoStruct sTrnsnUnk;
+u16* D_801F6D0C;
+s32 gTrnsnUnkState;
+VisMono sVisMono;
+Color_RGBA8_u32 gVisMonoColor;
+Struct_80140E80 D_801F6D38;
+Struct_80140E80* D_801F6D4C;
+HiresoStruct sHireso;
+u8 sBombersNotebookOpen;
+u8 sMotionBlurStatus;
+
+typedef enum {
+    /* 0 */ MOTION_BLUR_STOPPED,
+    /* 1 */ MOTION_BLUR_INITALIZING,
+    /* 2 */ MOTION_BLUR_RUNNING
+} MotionBlurStatus;
+
+void Play_DrawMotionBlur(PlayState* this) {
+    GraphicsContext* gfxCtx = this->state.gfxCtx;
+    s32 alpha;
+    Gfx* gfx;
+    Gfx* dlistHead;
+
+    if (R_MOTION_BLUR_PRIORITY_ENABLED) {
+        alpha = R_MOTION_BLUR_PRIORITY_ALPHA;
+
+        if (sMotionBlurStatus == MOTION_BLUR_STOPPED) {
+            sMotionBlurStatus = MOTION_BLUR_INITALIZING;
+        }
+    } else if (R_MOTION_BLUR_ENABLED) {
+        alpha = R_MOTION_BLUR_ALPHA;
+
+        if (sMotionBlurStatus == MOTION_BLUR_STOPPED) {
+            sMotionBlurStatus = MOTION_BLUR_INITALIZING;
+        }
+    } else {
+        alpha = 0;
+        sMotionBlurStatus = MOTION_BLUR_STOPPED;
+    }
+
+    if (sMotionBlurStatus != MOTION_BLUR_STOPPED) {
+        OPEN_DISPS(gfxCtx);
+
+        dlistHead = POLY_OPA_DISP;
+        gfx = Graph_GfxPlusOne(dlistHead);
+
+        gSPDisplayList(OVERLAY_DISP++, gfx);
+
+        this->pauseBgPreRender.fbuf = gfxCtx->curFrameBuffer;
+        this->pauseBgPreRender.fbufSave = this->unk_18E64;
+
+        if (sMotionBlurStatus == MOTION_BLUR_RUNNING) {
+            func_80170AE0(&this->pauseBgPreRender, &gfx, alpha);
+        } else {
+            sMotionBlurStatus = MOTION_BLUR_RUNNING;
+        }
+
+        func_801705B4(&this->pauseBgPreRender, &gfx);
+
+        gSPEndDisplayList(gfx++);
+
+        Graph_BranchDlist(dlistHead, gfx);
+
+        POLY_OPA_DISP = gfx;
+
+        CLOSE_DISPS(gfxCtx);
+    }
+}
+
+void Play_InitMotionBlur(void) {
+    R_MOTION_BLUR_ENABLED = false;
+    R_MOTION_BLUR_PRIORITY_ENABLED = false;
+    sMotionBlurStatus = MOTION_BLUR_STOPPED;
+}
+
+void Play_DestroyMotionBlur(void) {
+    R_MOTION_BLUR_ENABLED = false;
+    R_MOTION_BLUR_PRIORITY_ENABLED = false;
+    sMotionBlurStatus = MOTION_BLUR_STOPPED;
+}
+
+void Play_SetMotionBlurAlpha(u32 alpha) {
+    R_MOTION_BLUR_ALPHA = alpha;
+}
+
+void Play_EnableMotionBlur(u32 alpha) {
+    R_MOTION_BLUR_ALPHA = alpha;
+    R_MOTION_BLUR_ENABLED = true;
+}
+
+void Play_DisableMotionBlur(void) {
+    R_MOTION_BLUR_ENABLED = false;
+}
+
+void Play_ConvertBufferToI(void* destI, u16* srcBuf, s32 bufWidth, s32 pixelLeft, s32 pixelTop, s32 pixelRight,
```

Comment:
This is a bit confusing as a name if you don't know it's dealing with "Intensity images", and it doesn't mention the format of the input buffer which is expected to be RGBA16 only. I suggest something more like `Play_ConvertImageRgba16ToIntensity`, perhaps shuffling `Image` elsewhere